util/topgen/intermodule.py

```diff
@@ -98,6 +107,32 @@ def elab_intermodule(topcfg: OrderedDict):
                                              req_signal)
 
         rsp_len = len(rsps)
+        # decide signal format based on the `key`
+        sig_name = intersignal_format(req_struct)
+        req_struct["top_signame"] = sig_name
+
+        # Find package in req, rsps
+        if "package" in req_struct:
+            package = req_struct["package"]
+        else:
+            for rsp in rsps:
+                rsp_struct = find_intermodule_signal(list_of_intersignals,
+                                                     rsp_module, rsp_signal)
+                if "package" in rsp_struct:
+                    package = rsp_struct["package"]
+                    break
+            if not package:
+                package = ""
+
+        # Add to definition
+        definitions.append(
+            OrderedDict([('package', package),
+                         ('struct', req_struct["struct"]),
+                         ('signame', sig_name), ('width', req_struct["width"]),
+                         ('type', req_struct["type"])]))
+
+        req_struct["index"] = -1
+
         for i, rsp in enumerate(rsps):
             assert i == 0, "Handling multiple connections (array) isn't yet supported"
```

Comment:
can this be removed now..?

Reply:
It is more like connecting array of struct to array of struct. Which still is not supported...

Reply:
ooo i see what you're saing.